src/org/labkey/test/tests/SampleTypeLineageTest.java

```diff
@@ -0,0 +1,1088 @@
+
+        log("Make the parent sample the same sample that was just created.");
+        parentSample = newSampleName;
+        log(String.format("Using sample named '%s' as the parent.", parentSample));
+
+        waitAndClickAndWait(Locator.linkWithText(parentSample));
+
+        waitAndClickAndWait(Locator.linkWithText("derive samples from this sample"));
+
+        log("Again just create a single derived sample.");
+        waitAndClickAndWait(Locator.lkButton("Next"));
+        nameTxtbox = Locator.inputByNameContaining("_Name");
+        waitForElement(nameTxtbox);
+
+        newSampleName = namePrefix + newSampleIndex;
+        log(String.format("The new sample will be named '%s' and this one was a failure to launch and never left home.", newSampleName));
+
+        setFormElement(nameTxtbox, newSampleName);
+        clickAndWait(Locator.lkButton("Submit"));
+
+        List<String> newParents = new ArrayList<>();
+        newParents.add(parentSample);
+        newParents.addAll(parents);
+
+        log(String.format("Check that all samples '%s' are listed as parents.", newParents));
+        checkRowsInDataRegion("parentMaterials", "Name", newParents);
+
+        log("Also check that the run column for the parents are as expected.");
+        checkRowsInDataRegion("parentMaterials", "Run", List.of(runTextUI, " ", " ", " "));
+
+        log(String.format("Go to the 'derived' parent '%s' and verify it has '%s' as a child.", parentSample, newSampleName));
+
+        clickAndWait(Locator.linkWithText(parentSample));
+        checkRowsInDataRegion("childMaterials", "Name", List.of(newSampleName));
+
+        log(String.format("Verify again that '%s' parents are still as expected.", parentSample));
+        checkRowsInDataRegion("parentMaterials", "Name", parents);
+
+        String grandParent = parents.get(0);
+        log(String.format("Select one of '%s' parents, '%s' and verify that it now has two children.", parentSample, grandParent));
+
+        clickAndWait(Locator.linkWithText(grandParent));
+        checkRowsInDataRegion("childMaterials", "Name", Arrays.asList(newSampleName, parentSample));
+
+        log("Test complete.");
+    }
+
+    private void checkRowsInDataRegion(String dataRegionName, String columnName, List<String> expectedValues)
+    {
+        DataRegionTable dataRegionTable = new DataRegionTable(dataRegionName, this);
+
+        List<String> dataInTable = dataRegionTable.getColumnDataAsText(columnName);
+
+        checker().verifyEquals("Number of entries in the column '" + columnName + "' is not as expected.",
+                expectedValues.size(), dataInTable.size());
+
+        for(String expectedValue : expectedValues)
+        {
+            checker().verifyTrue(String.format("Value '%s' was not shown in column '%s' in data region '%s'.",
+                    expectedValue, columnName, dataRegionName),
+                    dataInTable.contains(expectedValue));
+
+        }
```

Comment:
Consider using `verifyEquals`. It could catch some errors that this would miss (wrong numbers of duplicated values).
```suggestion
        Collections.sort(expectedValues);
        Collections.sort(dataInTable);
        checker().verifyEquals("Entries in the column '" + columnName + "' are not as expected.",
                expectedValues, dataInTable);
```